qdrant-landing/content/articles/guide-filtering.md

```diff
@@ -0,0 +1,671 @@
+
+```csharp
+using Qdrant.Client;
+using static Qdrant.Client.Grpc.Conditions;
+
+var client = new QdrantClient("localhost", 6334);
+
+await client.ScrollAsync(
+	collectionName: "{collection_name}",
+	filter: Nested("diet", MatchKeyword("food", "meat") & Match("likes", true))
+);
+```
+
+The matching logic is adjusted to operate at the level of individual elements within an array in the payload.
+
+Nested filters function as though each element of the array is evaluated separately. The parent document will be considered a match if at least one array element satisfies the nested filter conditions.
+
+## Other creative uses for filters
+
+You can use filters to retrieve data points without knowing their `id`. You can search through data and manage it, solely by using filters. Let's take a look at some creative uses for filters:
+
+| Action | Description | Action | Description |
+|--------|-------------|--------|-------------|
+| [Delete Points](/documentation/concepts/points/#delete-points) | Deletes all points matching the filter. | [Set Payload](/documentation/concepts/payload/#set-payload) | Adds payload fields to all points matching the filter. |
+| [Scroll Points](/documentation/concepts/points/#scroll-points) | Lists all points matching the filter. | [Update Payload](/documentation/concepts/payload/#overwrite-payload) | Updates payload fields for points matching the filter. |
+| [Order Points](/documentation/concepts/points/#order-points-by-payload-key) | Lists all points, sorted by the filter. | [Delete Payload](/documentation/concepts/payload/#delete-payload-keys) | Deletes fields for points matching the filter. |
+| [Count Points](/documentation/concepts/points/#counting-points) | Totals the points matching the filter. | | |
+
+### Scrolling instead of searching
+
+In Qdrant, scrolling is used to iteratively retrieve large sets of points from a collection. It is particularly useful when you’re dealing with a large number of points and don’t want to load them all at once. Instead, Qdrant provides a way to scroll through the points in chunks or batches.
+
+You start by sending a scroll request to Qdrant with specific conditions like filtering by payload, vector search, or other criteria.
+
+```http
+POST /collections/{collection_name}/points/scroll
+{
+    "filter": {
+        "must": [
+            {
+                "key": "color",
+                "match": {
+                    "value": "red"
+                }
+            }
+        ]
+    },
+    "limit": 1,
+    "with_payload": true,
+    "with_vector": false
+}
+```
+The response contains a batch of points that match the criteria and a reference (offset or next page token) to retrieve the next set of points.
+
+Scrolling is designed to be efficient, especially when working with large datasets, because it minimizes the load on the server and reduces memory consumption on the client side by returning only manageable chunks of data at a time.
+
+## Filtering with the payload index
```

Comment:
This section is too late in the document (my impression), and these indices are needed for hnsw construction. Maybe we should put this right after the Basic Filtering Example section